### DISCUSSION TITLE
Error to deploy on Vercel

### DISCUSSION THREAD

**Discussion by avfigueredo** [statamic/ssg#102] [Q&A]
Hi, I'm trying to deploy on Vercel, but the error below it's occurring, I've followed the instructions on readme and added the ./build.sh file. 

Someone can help me to understand what's happening?

<img width="639" alt="Screen Shot 2022-06-22 at 14 20 38" src="https://user-images.githubusercontent.com/8737136/175039282-736176ba-9c9b-4568-85e4-285d2f209b60.png">

<img width="598" alt="Screen Shot 2022-06-22 at 14 14 08" src="https://user-images.githubusercontent.com/8737136/175038875-0fa95b53-6908-468c-be8d-8fc2358182ed.png">

**Comment by duncanmcclean** [comment]
Looks like you're needing to run `composer install` during your deployment pipeline.

**Reply by avfigueredo** [reply]
Yes, I'm using this script provided by the package and I set it to run when the project is building. 

`
#!/bin/sh

// Install PHP & WGET
yum install -y amazon-linux-extras
amazon-linux-extras enable php7.4
yum clean metadata
yum install php php-{common,curl,mbstring,gd,gettext,bcmath,json,xml,fpm,intl,zip,imap}
yum install wget

// INSTALL COMPOSER
EXPECTED_CHECKSUM="$(wget -q -O - https://composer.github.io/installer.sig)"
php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
ACTUAL_CHECKSUM="$(php -r "echo hash_file('sha384', 'composer-setup.php');")"

if [ "$EXPECTED_CHECKSUM" != "$ACTUAL_CHECKSUM" ]
then
    >&2 echo 'ERROR: Invalid installer checksum'
    rm composer-setup.php
    exit 1
fi

php composer-setup.php --quiet
rm composer-setup.php

// INSTALL COMPOSER DEPENDENCIES
php composer.phar install

// GENERATE APP KEY
php artisan key:generate

// BUILD STATIC SITE
php please ssg:generate

`

**Reply by elfedeb** [reply]
Had the same issue.
In my case it was that I was using PHP 8.1 locally and Vercel uses PHP 8.0 that's why the packages were doing some crazy stuff. 

Here are the steps I took to solve it.
- Check my local PHP version. Vercel uses 8.0 I was in PHP 8.1
- Updated my php version Mac via brew
- `brew list | grep php` get the php versions installed locally
- `brew tap shivammathur/php` get the php cask
- `brew install shivammathur/php/php@8.0`get the php8.0
- `brew unlink php` unlink the current php version
- `brew link php@8.0` use the 8.0 version
- `composer update` rebuild the composer.lock with the right versions
- `composer install` have statamic working back again

High-five yourself, you did it! ✋🤜🤛
After that I just pushed to github and worked perfectly.

Hope it helps :)
